telematic_system/historical_data_processing/rosbag2_processing_service/rosbag2_processing_service/service_manager.py

```diff
@@ -0,0 +1,214 @@
+        self.nc = NATS()
+        # List of rosbags to be processed
+        self.rosbag_queue = []
+
+        # Load config parameters
+        # Configured directory to read rosbags from
+        self.s3_mounted_dir = os.getenv("S3_MOUNTED_DIR")
+        # List of topics to be excluded from reading
+        self.topic_exclusion_list = os.getenv("TOPIC_EXCLUSION_LIST")
+        #NATS params
+        self.nats_ip_port = os.getenv("NATS_SERVER_IP_PORT")
+        #Logging configuration parameters
+        self.log_level = os.getenv("LOG_LEVEL")
+        self.log_name = os.getenv("LOG_NAME")
+        self.log_path = os.getenv("LOG_PATH")
+        self.log_rotation = int(os.getenv("LOG_ROTATION_SIZE_BYTES"))
+        self.log_handler_type = os.getenv("LOG_HANDLER_TYPE")
+
+        # Create logger
+        if self.log_handler_type == LogType.ALL.value:
+            # If all create log handler for both file and console
+            self.createLogger(LogType.FILE.value)
+            self.createLogger(LogType.CONSOLE.value)
+        elif self.log_handler_type == LogType.FILE.value or self.log_handler_type == LogType.CONSOLE.value:
+            self.createLogger(self.log_handler_type)
+        else:
+            self.createLogger(LogType.CONSOLE.value)
+            self.logger.warn("Incorrect Log type defined, defaulting to console")
+
+        # Load Influx params
+        self.influx_bucket = os.getenv("INFLUX_BUCKET")
+        self.influx_org = os.getenv("INFLUX_ORG")
+        self.influx_token = os.getenv("INFLUX_TOKEN")
+        self.influx_url = os.getenv("INFLUX_URL")
+
+        #Fields in the ros message to force to string type.
+        self.to_str_fields = os.getenv("TO_STR_FIELDS")
+        # Fields in the ros message to ignore
+        self.ignore_fields = os.getenv("IGNORE_FIELDS")
+
+        self.accepted_file_extensions = os.getenv("ACCEPTED_FILE_EXTENSIONS")
+
+        # Create rosbag parser object
+        self.rosbag_parser = Rosbag2Parser(self.influx_bucket, self.influx_org, self.influx_token, self.influx_url, self.topic_exclusion_list, self.s3_mounted_dir, self.to_str_fields, self.ignore_fields, self.logger, self.accepted_file_extensions)
+
+        #nats connection status
+        self.is_nats_connected = False
+
+        # Nats request topic
+        self.nats_request_topic = os.getenv("NATS_REQUEST_TOPIC")
+
+        #TODO: Add Mysql database connection
+        #self.db = _mysql.connect(host="", user="", password="", database="")
+
+    def createLogger(self, log_type):
+        """Creates log file for the ServiceManager with configuration items based on the settings input in the params.yaml file"""
+        # create log file and set log levels
+        self.logger = logging.getLogger(self.log_name)
+        now = datetime.now()
+        dt_string = now.strftime("_%m_%d_%Y_%H_%M_%S")
+        log_name = self.log_name + dt_string + ".log"
+        formatter = logging.Formatter(
```

Comment:
Can we add the threads name in the log format as well? ```%(asctime)s - %(threadName)s - %(name)s - %(funcName)s - %(levelname)s - %(message)s```

Reply:
I tried testing with this added, but its not giving any usable information. We are printing the function name where the log is from and only the nats process is being run on a separate thread